toucan_connectors/mongo/mongo_connector.py

```diff
@@ -242,13 +242,14 @@ def get_slice(
             df, stats=DataStats(total_returned_rows=total_count, total_rows=total_count)
         )
 
-    def get_df_with_regex(
+    def get_slice_with_regex(
         self,
         data_source: MongoDataSource,
-        field: str,
+        fields: List[str],
         regex: Pattern,
         permissions: Optional[str] = None,
         limit: Optional[int] = None,
+        offset: Optional[int] = None,
     ) -> pd.DataFrame:
```

Comment:
```suggestion
    ) -> DataSlice:
```

You're not returning the `.df` attribute anymore